STY: Tag update (#996)
pysat/pysat: CHANGELOG.md

   * Added docstring linting via flake-docstrings and hacking to CI tests
   * Added clarification to the tutorials
   * Added limitations to the `listify` docstring

>    * Updated `tag` and `inst_id` keyword docstrings for test instruments,
>      as well as for `pysat.instruments.templates.template_instrument`, 
>      `pysat.instruments.pysat_netcdf`, and 
>      `pysat.instruments.methods.general.list_files`

Reviewer: ```suggestion
   * Updated `tag` and `inst_id` keywords and docstrings to new standards```

   * Updated pandas syntax to be compatible with pandas 2.0 (pandas 1.4
     deprecations)
   * Cleaned up excess variables upon import

>    * Removed data_path check from `pysat.instruments.methods.general.list_files`

Reviewer: ```suggestion
   * Removed `data_path` check from `pysat.instruments.methods.general.list_files`
```
